### PR TITLE
[bluetooth.govee] Govee Bluetooth Binding initial contribution

### PR DESCRIPTION
**Pull request by cpmeister**
I got some of these devices in order to try to test the new bluetooth dynamic channel generation feature I've been working on. Sadly they didn't really help, but since I already had the devices on my hands I made a quick binding for them anyway.

Signed-off-by: Connor Petty <mistercpp2000+gitsignoff@gmail.com>

### CODE REVIEW

**Comment by cpmeister** [comment]
If you consider this a test of the new binding generation script, it seems to have really botched the `bundles/pom.xml` file by adding blank lines everywhere.

File: bundles/org.openhab.binding.bluetooth.govee/README.md
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/README.md:16]: What do you mean by "*two* thing types"?

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeDiscoveryParticipant.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeDiscoveryParticipant.java:39]:
```suggestion
@Component(service = BluetoothDiscoveryParticipant.class)
```

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml:18]: Is this Channel used? Same for below.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml:31]: You could define a config reference to get rid of the redundant config-descriptions.

**Review by fwolter — CHANGES_REQUESTED** [review]

**Comment by cpmeister** [comment]
So my initial implementation of this binding didn't actually work when I tested it, so I've had to rewrite it from scratch.
Been working and testing this for the past month and I can finally say this is ready for review.

File: bundles/org.openhab.binding.bluetooth.govee/README.md
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/README.md:29]: Unit

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java:145]: What's the purpose of this?

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetTemHumCommand.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetTemHumCommand.java:33]: Can this be removed?

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml:181]: Strange line break.

File: bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java:40]: I like this. Love to see this at some more prominent location.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java:75]:
```suggestion
    private static final byte[] SCAN_HEADER = { (byte) 0xFF, (byte) 0x88, (byte) 0xEC };
```

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetCommand.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetCommand.java:21]:
```suggestion
@NonNullByDefault
public abstract class GetCommand extends GoveeCommand {
```

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/SetCommand.java
Comment (fwolter) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/SetCommand.java:19]:
```suggestion
@NonNullByDefault
public abstract class SetCommand extends GoveeCommand {
```

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.bluetooth.govee/README.md
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/README.md:26]: The asterisk means that the config is available to both `goveeHydrometer` and the `goveeHydrometerMonitor` thing types. I'll make it more explicit in the readme.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java:145]: wow, I totally forgot to remove that. It was strictly for debugging.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml:181]: blame spotless

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/ConnectedBluetoothHandler.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/ConnectedBluetoothHandler.java]: done

File: bundles/org.openhab.binding.bluetooth.govee/README.md
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/README.md]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/ConnectedBluetoothHandler.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/ConnectedBluetoothHandler.java]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/ConnectedBluetoothHandler.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/ConnectedBluetoothHandler.java]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml]: tried my best, but there is definitely some kinds of spotless formatting bug at play.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetCommand.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetCommand.java:23]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/SetCommand.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/SetCommand.java:22]: done

File: bundles/org.openhab.binding.bluetooth.govee/README.md
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/README.md]: added

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetTemHumCommand.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetTemHumCommand.java]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/feature/feature.xml
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/feature/feature.xml:19]: That is kinda the case for all of the bluetooth bindings...
"extension" refers to the fact that the binding adds functionality to the base bluetooth bindings.
"adapter"  is used the pom.xml as part of the project name. But strictly speaking I don't really think this is appropriate and the word should probably be changed to `extension` instead. But this standard was in place for all of the other bluetooth bindings so if I'm going to change it it should be a separate PR. 

I do agree with you that these naming conventions are a bit confusing at this point and I would love to change them.

**Review by fwolter — APPROVED** [review]
LGTM

**Comment by cpmeister** [comment]
@openhab/add-ons-maintainers This needs a second reviewer if someone can take a look. Thanks.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Hilbrand) [inline bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml:48]: element `required` is deprecated. Use attribute. But in this case as it's the same as the default value you can just remove it. Same for other parameter.

File: bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java
Comment (Hilbrand) [inline bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java:94]: Maybe not use apache `StringUtils` as we want to get ride of it.

**Review by Hilbrand — CHANGES_REQUESTED** [review]
Just some small stuff.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/feature/feature.xml
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/feature/feature.xml]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetOrSetHumCaliCommand.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/command/hygrometer/GetOrSetHumCaliCommand.java]: Thanks for finding that. Removed.

File: bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/main/resources/OH-INF/thing/thing-types.xml]: done

File: bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java
Comment (cpmeister) [reply bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java]: done

**Review by Hilbrand — APPROVED** [review]
LGTM

**Pull request merged**
